src/JsonWriter.cpp

```diff
@@ -101,48 +101,55 @@ void JsonWriteValueString(JsonWriter *writer, const char *value, size_t maxLen)
     if (writer->m_PrependComma)
         JsonWriteChar(writer, ',');
 
-    JsonWriteChar(writer, '"');
-
-    size_t len = 0;
-    while (*value != 0 && len < maxLen)
+    if (value == nullptr)
     {
-        char ch = *(value++);
-        if (ch == '"')
-        {
-            JsonWrite(writer, "\\\"", 2);
-        }
-        else if (ch == '\\')
-        {
-            JsonWrite(writer, "\\\\", 2);
-        }
-        else if (ch == 0x0A)
-        {
-            JsonWrite(writer, "\\n", 2);
-        }
-        else if (ch == 0x0D)
-        {
-            JsonWrite(writer, "\\r", 2);
-        }
-        else if (ch == 0x09)
-        {
-            JsonWrite(writer, "\\t", 2);
-        }
-        else if (ch == 0x0C)
-        {
-            JsonWrite(writer, "\\f", 2);
-        }
-        else if (ch == 0x08)
-        {
-            JsonWrite(writer, "\\b", 2);
-        }
-        else
+        JsonWrite(writer, "null", 4);
+    }
+    else
+    {
+        JsonWriteChar(writer, '"');
+
+        size_t len = 0;
+        while (*value != 0 && len < maxLen)
         {
-            JsonWriteChar(writer, ch);
+            char ch = *(value++);
+            if (ch == '"')
```

Comment:
Please consider using a conversion table here to convert the characters instead of multiple if statements

Reply:
That's fine, but all I did in this PR was indent things - I am not going to do any further refactoring here now.